geoPipe.py

```diff
@@ -26,9 +30,121 @@
 pd.set_option('display.max_rows', 100000)
 pd.set_option('display.width', 1000)
 
+
+# CONFIGURATION VALUES
+
+MIN_PATH_DIST_METERS = 200
+MAX_REPAIR_DIST_METERS = 150
+BUS_RADIUS_METERS = 800
+LOOP_COMPLETION_THRESHOLD_METERS = 20
+POLYGON_SIMPLIFICATION_THRESHOLD = 0.0004
+
+SLEEP_BETWEEN_QUERIES_SEC = 4
+OFFSET_START = -1  # -1 default
+
+REGIONS = {
+	"Southeast Michigan": {
+		# Clinton, MI (SW corner) to Richmond, MI (NE corner)
+		"bounds": "-84.046297,42.042086,-82.757646,42.816828",
+		# Dearborn area (SW corner) to St. Claire Shores (NE corner)
+		# 	"bounds": "-83.280879,42.268401,-82.806811,42.659880",
+
+		# Should be the same as on TOTAGO (5 = SE Michigan / Detroit)
+		"region_id": 5 
+	}
+}
+TYPES = {
+	"bicycle": "bicycle",
+	"foot": "foot",
+	"park": "park"
+}
+
+### TO CHANGE MANUALLY ###
+REGION = REGIONS["Southeast Michigan"]
+TYPE = TYPES["foot"]
+WHERE_QUERY = "Park_Size_Acres>20"
+########################## 
+
+
+# History/notes:
+# 	WHERE_QUERY = "Park_Size_Acres>5+AND+Park_Size_Acres<10"
+# 	Test region queries here: 
+#		https://nominatim.openstreetmap.org/search.php
+
+# ArcGIS UI for API
+# https://server3.tplgis.org/arcgis3/rest/services/ParkServe/ParkServe_Shareable/MapServer/0/query?where=Park_Size_Acres+%3E+5+AND+Park_Size_Acres+%3C+10&text=&objectIds=&time=&geometry=-83.280879%2C42.268401%2C-82.806811%2C42.659880&geometryType=esriGeometryEnvelope&inSR=%7B%22wkid%22+%3A+4326%7D&spatialRel=esriSpatialRelEnvelopeIntersects&relationParam=&outFields=Park_Name%2CPark_Local_Owner%2CPark_Local_Manager%2CPark_Size_Acres&returnGeometry=true&returnTrueCurves=true&maxAllowableOffset=&geometryPrecision=&outSR=%7B%22wkid%22+%3A+4326%7D&having=&returnIdsOnly=false&returnCountOnly=false&orderByFields=&groupByFieldsForStatistics=&outStatistics=&returnZ=true&returnM=false&gdbVersion=&historicMoment=&returnDistinctValues=false&resultOffset=1&resultRecordCount=500&queryByDistance=&returnExtentOnly=false&datumTransformation=&parameterValues=&rangeValues=&quantizationParameters=&f=html
+
+#################
+
+# type must be "bicycle" or "foot"
+def getOSMQueryByPolygon(type, polygon, min_path_dist_meters):
+	if type == 'foot': 
+		return '[out:json][timeout:1000]; \
+			(\
+				way["highway"~"path|footway|footpath|bridleway"]\
+				["footway"!~"sidewalk|crossing"] \
+				["bicycle"!~"yes|designated"]\
+				(poly:"{}");\
+			);(._;>;);out;'.format(polygon)
+
+	if type == 'bicycle':
+		return '[out:json][timeout:25]; \
+			(\
+				way["highway"~"path|footway|footpath|bridleway"]\
+				["footway"!~"sidewalk|crossing"]\
+				["bicycle"~"yes|designated"]\
+				(poly:"{}");\
+				way["highway"~"cycleway"]\
+				(poly:"{}");\
+			);(._;>;);out;'.format(polygon, polygon)
+	
```

Comment:
Hmm so you're just manually formatting the JSON being returned here?: I haven't seen syntax like this.

Reply:
No, this is actually a query to OpenStreetMap database for trails. Then the response is set to come back as JSON. The .format is just a way to put variables into a Python string, nothing fancy, just the same as "str1 " + variable + " str2"